### ISSUE TITLE
Compile error both stand alone as within redox build

### ISSUE THREAD

**Issue by DaanHoogland** [briansmith/ring#415]
compiling as dependency of redox or stand alone i get the error:
```
...
error: entity not found
error: entity not found
make: *** No rule to make target '/home/dahn/prj/ring/target/debug/build/ring-27376952505f4099/out/obj/crypto/', needed by '/home/dahn/prj/ring/target/debug/build/ring-27376952505f4099/out/obj/crypto/aes/aes.o'.  Stop.
thread 'main' panicked at 'make execution failed', build.rs:236
```
this is on fedora 25 and, as mentioned, with nightly rustc dated the 12th to the 15th

Here's a link to the full error message on pastebin: http://pastebin.com/EH8R0Pyz

**Comment by DaanHoogland** [comment]
it took me a while to realise this can not possibly have anything to do  with rust nightly (2017-01-12 till 2016-01-15). It must be either cargo, cc or make;
```
[dahn@blimbing ~]$ cargo --version
cargo 0.15.0-nightly (298a012 2016-12-20)
[dahn@blimbing ~]$ cc --version
cc (GCC) 6.3.1 20161221 (Red Hat 6.3.1-1)
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[dahn@blimbing ~]$ make --version
GNU Make 4.1
Built for x86_64-redhat-linux-gnu
Copyright (C) 1988-2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by briansmith** [comment]
This looks like a makefile issue or similar. Let's revisit after #321 lands.

**Comment by briansmith** [comment]
@DaanHoogland Could you please try with the current git master? The build system has been completely rewritten. (Note that we might have broken Redox support accidentally in the process.)

**Comment by DaanHoogland** [comment]
thanks, I can confirm it builds standalone.

**Comment by briansmith** [comment]
Thanks @DaanHoogland

**Comment by DaanHoogland** [comment]
So redox breaks on ring, using 0.7.1. I'll see if updating gives relief (not your problem by any standard ;)

**Comment by DaanHoogland** [comment]
One more question though, is 0.7.1 the latest and when is a release with this fixed buildsystem planned? On my fedora machine redox won't build and I can't find a newer release on github.

**Comment by briansmith** [comment]
I just released 0.7.2 and 0.7.3 with the new build system. I'm not sure the redox build is working yet since I don't even know how to set up the build env. for redox.

**Comment by DaanHoogland** [comment]
thanks @briansmith I'll find some spare time to do an updated build (if needed @jackpot51 might have that automated ;)
